### ISSUE TITLE
Included files don't invalidate the cache (Parcel 2)

### ISSUE THREAD

**Issue by victorandree** [parcel-bundler/parcel#3708]
**Note:** Updated the issue with the latest alpha release (3.2). Included files are now "watched" but still don't cause cache invalidation – demonstrated by the reproduction repository (unless it's a problem with the pug transformer specifically).

# 🐛 bug report

In Parcel 2 (2.0.0-alpha.3.2), it doesn't seem like files included by a transform as an asset's `includedFiles` are considered for cache invalidation ~~or rebuilding (when watching).~~

This can be demonstrated by e.g. using the `include` feature of Pug, where the v2 transformer will loop over includes and do [`asset.addIncludedFile({filePath})`](https://github.com/parcel-bundler/parcel/blob/v2/packages/transformers/pug/src/PugTransformer.js#L36)

I initially reported this as a comment on the call to test Parcel 2, but decided to create a separate issue because I can't see it tracked already. Apologies if I've missed it.

## 🎛 Configuration (.babelrc, package.json, cli command)

I have a reproduction repository at https://github.com/victorandree/test-parcel2-includes where master is using v2, and a branch [parcel@v1](https://github.com/victorandree/test-parcel2-includes/tree/parcel%40v1) uses Parcel 1.

## 🤔 Expected Behavior

Since included files are merged into the asset, changes made to them should probably invalidate the asset cache ~~and trigger rebuilds~~.

## 😯 Current Behavior

~~In Parcel 2, when running `npx parcel watch`, changes made to an included file don't result in the assets being rebuilt.~~ When running `npx parcel build`, changes made to an included file aren't reflected (because of caching).

## 💁 Possible Solution

Treat included files as dependencies for the above purposes.

## 🔦 Context

I've been trying to write a transform for Parcel 2, where it's not appropriate to treat includes as dependencies – because the includes aren't assets on their own. I noticed that changes made to these includes had no effect on my main asset.

## 💻 Code Sample

I have a reproduction repository at https://github.com/victorandree/test-parcel2-includes where master is using v2, and a branch [parcel@v1](https://github.com/victorandree/test-parcel2-includes/tree/parcel%40v1) uses Parcel 1.

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.0.0-alpha.3.2
| Node             | 12.13.0
| npm/Yarn         | 6.12.0
| Operating System | macOS 10.14.16

**Comment by devongovett** [comment]
Looks like we are missing nodes for the included files being added here: https://github.com/parcel-bundler/parcel/blob/v2/packages/core/core/src/RequestGraph.js#L353. @padmaia is working on a refactor to the RequestGraph at the moment, so we'll fix this once that lands.

**Comment by victorandree** [comment]
For what it's worth, v2.0.0-alpha.3.2 seems to trigger rebuilds from included files but the cache is still not invalidated for the parent file.

**Comment by mischnic** [comment]
Update: seem like it triggers a rebuild but doesn't invalidate the cache (so it says "Bundling" and "Build in ..." but doesn't actually update the bundle)

**Comment by mischnic** [comment]
The problem is that the hash of included files isn't factored into the transformation cache key (that hash is actually never generated):

https://github.com/parcel-bundler/parcel/blob/293c9753847a7cba8eb35b32c3f01318ea5f3539/packages/core/core/src/Transformation.js#L362-L376
